lib/flop/builder.ex

```diff
@@ -30,20 +30,28 @@ defmodule Flop.Builder do
     :ilike_or
   ]
 
-  def filter(_, %Filter{field: nil}, c), do: c
-  def filter(_, %Filter{value: nil}, c), do: c
-
-  def filter(
-        schema_struct,
-        %Filter{field: field} = filter,
-        conditions
-      ) do
-    build_op(
-      conditions,
-      schema_struct,
-      get_field_type(schema_struct, field),
-      filter
-    )
+  def filter(query, schema_struct, filters, extra_opts) do
+    {query, conditions} =
+      filters
+      |> Enum.reduce({query, true}, fn
```

Comment:
I'd prefer the reducer function to be a named private function (`filter_reducer/2`).